File: Easy/esonero/meta_esonerodisabilview/meta_esonerodisabilview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_esonerodisabilview
{
    public class Meta_esonerodisabilview : Meta_easydata 
	{
        public Meta_esonerodisabilview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "esonerodisabilview") {
				Name = "Definizione degli esoneri di invalidità";
			EditTypes.Add("disabil");
			ListingTypes.Add("disabil");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idesonero"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "disabil": {
						DescribeAColumn(T, "esonero_aa", "Anno accademico", nPos++);
						DescribeAColumn(T, "title", "Denominazione", nPos++);
						DescribeAColumn(T, "esonero_description", "Descrizione", nPos++);
						DescribeAColumn(T, "esonero_applunavolta", "Applicabile una sola volta", nPos++);
						DescribeAColumn(T, "costoscontodef_title", "Sconto", nPos++);
						DescribeAColumn(T, "esoneroanskind_title", "Tipologia Codice ANS", nPos++);
						DescribeAColumn(T, "esoneroanskind_description", "Descrizione Codice ANS", nPos++);
						DescribeAColumn(T, "esonero_retroattivo", "Retroattivo", nPos++);
						DescribeAColumn(T, "esonero_soloconisee", "Applicabile solo con ISEE", nPos++);
						DescribeAColumn(T, "esonero_disabil_percmax", "Percentuale massima", nPos++);
						if (T.Columns.Contains("esonero_disabil_percmax")) T.Columns["esonero_disabil_percmax"].ExtendedProperties["format"] = "fixed.2";
						DescribeAColumn(T, "esonero_disabil_percmin", "Percentuale minima", nPos++);
						if (T.Columns.Contains("esonero_disabil_percmin")) T.Columns["esonero_disabil_percmin"].ExtendedProperties["format"] = "fixed.2";
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "disabil": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/esonero/meta_esonerodisabilview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_esonerodisabilview() {
        MetaData.apply(this, ["esonerodisabilview"]);
        this.name = 'meta_esonerodisabilview';
    }

    meta_esonerodisabilview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_esonerodisabilview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'disabil':
						this.describeAColumn(table, 'aa', 'Anno accademico', null, 1000, 9);
						this.describeAColumn(table, 'esonero_disabil_percmax', 'Percentuale massima', 'fixed.2', 2000, null);
						this.describeAColumn(table, 'title', 'Denominazione', null, 3000, 50);
						this.describeAColumn(table, 'esonero_disabil_percmin', 'Percentuale minima', 'fixed.2', 3000, null);
						this.describeAColumn(table, 'esonero_description', 'Descrizione', null, 4000, 256);
						this.describeAColumn(table, 'esonero_applunavolta', 'Applicabile una sola volta', null, 5000, null);
						this.describeAColumn(table, 'costoscontodef_title', 'Sconto', null, 6200, 2024);
						this.describeAColumn(table, 'esoneroanskind_title', 'Tipologia Codice ANS', null, 7200, 50);
						this.describeAColumn(table, 'esoneroanskind_description', 'Descrizione Codice ANS', null, 7300, 256);
						this.describeAColumn(table, 'esonero_retroattivo', 'Retroattivo', null, 8000, null);
						this.describeAColumn(table, 'esonero_soloconisee', 'Applicabile solo con ISEE', null, 9000, null);
//$objCalcFieldConfig_disabil$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idesonero"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "disabil": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('esonerodisabilview', new meta_esonerodisabilview('esonerodisabilview'));

	}());
